samples/bluetooth/central_smp_client/src/main.c

```diff
@@ -287,29 +300,51 @@ static void smp_echo_rsp_proc(struct bt_dfu_smp *dfu_smp)
 		size_t payload_len = ((uint16_t)smp_rsp_buff.header.len_h8) << 8 |
 				      smp_rsp_buff.header.len_l8;
 
-		CborError cbor_error;
-		CborParser parser;
-		CborValue value;
-		struct cbor_buf_reader reader;
+		zcbor_state_t zsd[CBOR_DECODER_STATE_NUM];
+		struct zcbor_string value = {0};
+		char map_key[SMP_ECHO_MAP_KEY_MAX_LEN];
+		char map_value[SMP_ECHO_MAP_VALUE_MAX_LEN];
+		bool ok;
 
-		cbor_buf_reader_init(&reader, smp_rsp_buff.payload, payload_len);
+		zcbor_new_decode_state(zsd, ARRAY_SIZE(zsd), smp_rsp_buff.payload, payload_len, 1);
 
-		cbor_error = cbor_parser_init(&reader.r, 0, &parser, &value);
+		/* Stop decoding on the error. */
+		zsd->constant_state->stop_on_error = true;
 
-		if (cbor_error != CborNoError) {
-			printk("CBOR parser initialization failed (err: %d)\n",
-			       cbor_error);
+		zcbor_map_start_decode(zsd);
+		ok = zcbor_tstr_decode(zsd, &value);
+
+		if (!ok && (value.len != 1)) {
```

Comment:
`!ok ||  (value.len != 1) || *value.value != 'd'`

And then you can drop the `map_key` completely.